code/modules/mob/living/carbon/xenomorph/castes/gorger/abilities_gorger.dm

```diff
@@ -0,0 +1,188 @@
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/carbon/xenomorph/gorger/X = owner
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat != DEAD)
+			to_chat(X, span_notice("We can only use still, dead hosts to heal."))
+			return FALSE
+		else if(get_dist(X, H) > 1)
+			to_chat(X, span_notice("We need to be next to our meal."))
+			return FALSE
+		return TRUE
+	if(isxeno(target))
+		if(X == target)
+			if(!COOLDOWN_CHECK(X, rejuvenate_self_cooldown))
+				to_chat(X, span_notice("We need another [round(COOLDOWN_TIMELEFT(X, rejuvenate_self_cooldown) / 10)] seconds before we can revitalize ourselves."))
+				return FALSE
+			if(X.blood_bank < 10)
+				to_chat(X, span_notice("We need [10 - X.blood_bank]u more blood to revitalize ourselves."))
+				return FALSE
+		else
+			if(!X.line_of_sight(target) || get_dist(X, target) > 2)
+				to_chat(X, span_notice("It is beyond our reach, we must be close and our way clear."))
+				return FALSE
+			if(X.blood_bank < 5)
+				to_chat(X, span_notice("We need [5 - X.blood_bank]u more blood to restore a sister."))
+				return FALSE
+			if(isdead(target))
+				to_chat(X, span_notice("We can only help living sisters."))
+				return FALSE
+		return TRUE
+	to_chat(X, span_notice("We can only drain or restore familiar biological lifeforms."))
+	return FALSE
+
+/datum/action/xeno_action/activable/rejuvenate/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/gorger/X = owner
+	if(ishuman(A))
+		var/mob/living/carbon/human/target = A
+		while(X.health < X.maxHealth && do_after(X, 2 SECONDS, TRUE, A, BUSY_ICON_HOSTILE))
+			X.heal_wounds(1.1, TRUE)
+			X.adjust_sunder(-1.5)
+			target.blood_volume -= 2
+		to_chat(X, span_notice("We feel fully restored."))
+	else if(A == X)
+		var/mob/living/carbon/xenomorph/target = A
+		X.blood_bank -= 10
+		to_chat(X, span_notice("Blood bank: [X.blood_bank]%"))
+		to_chat(X, span_notice("We tap into our reserves for nourishment."))
+		target.apply_status_effect(/datum/status_effect/xeno_rejuvenate, 5 SECONDS)
+		COOLDOWN_START(X, rejuvenate_self_cooldown, 20 SECONDS)
+	else
+		var/mob/living/carbon/xenomorph/target = A
+		if(!do_mob(X, A, 1 SECONDS, BUSY_ICON_FRIENDLY, BUSY_ICON_MEDICAL))
+			return
+		if(!X.line_of_sight(target) || get_dist(X, target) > 2)
+			to_chat(X, span_notice("Beyond our reach..."))
+			return
+		target.adjustBruteLoss(-X.maxHealth*0.3)
+		target.adjustFireLoss(-X.maxHealth*0.3)
+		X.blood_bank -= 5
+		to_chat(X, span_notice("Blood bank: [X.blood_bank]%"))
+	succeed_activate()
+	add_cooldown()
+
+// ***************************************
+// *********** Carnage
+// ***************************************
+/datum/action/xeno_action/activable/carnage
+	name = "Carnage"
+	action_icon_state = "Carnage"
+	mechanics_text = ""
+	ability_name = "carnage"
+	cooldown_timer = 40 SECONDS
+	plasma_cost = 100
+
+/datum/action/xeno_action/activable/carnage/can_use_ability(atom/A, silent = FALSE, override_flags)
```

Comment:
does... this do anything?

Reply:
removed plasma cost, the ability applies a buff and that's it